benchmarks/src/main/java/org/elasticsearch/benchmark/script/ScriptScoreBenchmark.java

```diff
@@ -0,0 +1,208 @@
+    private final PluginsService pluginsService = new PluginsService(
+        Settings.EMPTY,
+        null,
+        null,
+        Path.of(System.getProperty("plugins.dir")),
+        List.of()
+    );
+    private final ScriptModule scriptModule = new ScriptModule(Settings.EMPTY, pluginsService.filterPlugins(ScriptPlugin.class));
+
+    private final Map<String, MappedFieldType> fieldTypes = Map.ofEntries(
+        Map.entry("n", new NumberFieldType("n", NumberType.LONG, false, false, true, true, null, Map.of()))
+    );
+    private final IndexFieldDataCache fieldDataCache = new IndexFieldDataCache.None();
+    private final CircuitBreakerService breakerService = new NoneCircuitBreakerService();
+    private SearchLookup lookup = new SearchLookup(
+        fieldTypes::get,
+        (mft, lookup) -> mft.fielddataBuilder("test", lookup).build(fieldDataCache, breakerService)
+    );
+
+    @Param({ "expression", "metal", "painless_cast", "painless_def" })
+    private String script;
+
+    @Param({ "16" })
+    private double indexingBufferMb;
+
+    private ScoreScript.Factory factory;
+
+    private IndexReader reader;
+
+    @Setup
+    public void setupScript() {
+        switch (script) {
+            case "expression":
+                factory = scriptModule.engines.get("expression").compile("test", "doc['n'].value", ScoreScript.CONTEXT, Map.of());
+                break;
+            case "metal":
+                factory = bareMetalScript();
+                break;
+            case "painless_cast":
+                factory = scriptModule.engines.get("painless")
+                    .compile(
+                        "test",
+                        "((org.elasticsearch.index.fielddata.ScriptDocValues.Longs)doc['n']).value",
+                        ScoreScript.CONTEXT,
+                        Map.of()
+                    );
+                break;
+            case "painless_def":
+                factory = scriptModule.engines.get("painless").compile("test", "doc['n'].value", ScoreScript.CONTEXT, Map.of());
+                break;
+            default:
+                throw new IllegalArgumentException("Don't know how to implement script [" + script + "]");
+        }
+    }
+
+    @Setup
+    public void setupIndex() throws IOException {
+        Path path = Path.of(System.getProperty("tests.index"));
+        IOUtils.rm(path);
+        Directory directory = new MMapDirectory(path);
+        try (
+            IndexWriter w = new IndexWriter(
+                directory,
+                new IndexWriterConfig().setOpenMode(OpenMode.CREATE).setRAMBufferSizeMB(indexingBufferMb)
+            )
+        ) {
+            for (int i = 1; i <= 1_000_000; i++) {
+                w.addDocument(List.of(new SortedNumericDocValuesField("n", i)));
+            }
+            w.commit();
+        }
+        reader = DirectoryReader.open(directory);
+    }
+
+    @Benchmark
+    public void benchmark() throws IOException {
```

Comment:
It's a best practice in JMH to either return a value (in your case `topDocs`) or to use `Blackhole` to consume the value in order to avoid / reduce the potential for dead code eliminitation. Therefore I propose you return `topDocs` here. JMH will then consume the return value appropriately.